### PR TITLE
Add DFG-to-FOS mappings utility function

### PR DESCRIPTION
**Pull request by jrhoads**
## Purpose
Convert between dfg subject vocabulary and Field of study vocabulary

closes: datacite/datacite#1586

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)

- [x] New feature (non-breaking change which adds functionality)

- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Reviewer, please remember our [guidelines](https://datacite.atlassian.net/wiki/spaces/TEC/pages/1168375809/Pull+Request+Guidelines):

- Be humble in the language and feedback you give, ask don't tell.
- Consider using positive language as opposed to neutral when offering feedback. This is to avoid the negative bias that can occur with neutral language appearing negative.
- Offer suggestions on how to improve code e.g. simplification or expanding clarity.
- Ensure you give reasons for the changes you are proposing.

### CODE REVIEW

File: lib/bolognese/utils.rb [hunk 1]
```diff
@@ -1369,5 +1373,20 @@ def hsh_to_fos(hsh)
           "lang" => hsh["lang"] }.compact]
       end
     end
+
+    def dfg_ids_to_fos(dfg_ids)
+      dfgs = JSON.load(File.read(File.expand_path('../../../resources/oecd/dfg-mappings.json', __FILE__))).fetch("dfgFields")
```
Comment (digitaldogsbody) [inline lib/bolognese/utils.rb:1378]: I wonder if it's worth defining a constant for the resources directory and using that, to make the directory path a little cleaner (and potentially more portable if we ever need to move it?).

Similar to the spec fixtures:
https://github.com/datacite/bolognese/blob/master/spec/spec_helper.rb#L36-L38
https://github.com/datacite/bolognese/blob/master/spec/datacite_utils_spec.rb#L180

**Review by digitaldogsbody — APPROVED** [review]
Looks good to me :)

File: lib/bolognese/utils.rb (hunk 1 above)
Comment (jrhoads) [reply lib/bolognese/utils.rb:1378]: Can do

**Review by digitaldogsbody — APPROVED** [review]
Refactor looks great!

**Pull request merged**